### PR TITLE
PuzzleManager

### PR DESCRIPTION
**Pull request by Carlijavier5**
A few basic scripts intended for later use;

**PuzzleManager.cs:** Singleton class to record level completion using enums;
**PuzzleHandler.cs**: Abstract class with code to declare a puzzle "Complete;"
**PuzzleProc.cs:** Sample class to initiate Puzzle Scenes based on completion;
**TransitionManager.cs:** Singleton class to handle screen fade and scene transitions;
_\+ Some prefabs for the two singletons, and a few random testing scripts + 2 testing scenes under my folder;_

Note that most of the Puzzle Management code is wrapped around the **PuzzleManagement namespace**;
Also note that the logic in my scenes won't work if **they aren't listed properly under Build Settings**;

**Disclaimer:** I did not write any code pertaining the existing puzzles, their logic, nor the conditions under which they may declare themselves *complete*;

**TL;DR:** Merge my PR, thank you <3
- Carlos;

### CODE REVIEW

**Review by JungwooJang119 — APPROVED** [review]
wahooo!
